R/ttestonesample.R

```diff
@@ -237,61 +237,69 @@ TTestOneSample <- function(jaspResults, dataset = NULL, options, ...) {
 }
   
 .ttestOneSampleComputeMainTableRow <- function(variable, dataset, test, testStat, optionsList, options) {
-  direction <- switch(options$hypothesis,
+  direction <- switch(options[["hypothesis"]],
                       "notEqualToTestValue"  ="two.sided",
                       "greaterThanTestValue" ="greater",
                       "lessThanTestValue"    ="less")
   dat <- na.omit(dataset[[ .v(variable) ]])
   n   <- length(dat)
   if (test == "Wilcoxon") {
-    r <- stats::wilcox.test(dat, alternative = direction, mu = options$testValue,
-                            conf.level = optionsList$percentConfidenceMeanDiff, conf.int = TRUE)
-    df   <- ifelse(is.null(r$parameter), "", as.numeric(r$parameter))
+    # Note(Alexander): Effect size estimate of Wilcoxon is based on the centralised statistic
+    tempResult <- stats::wilcox.test(dat, "alternative" = direction, "mu" = options[["testValue"]],
+                                     "conf.level" = optionsList[["percentConfidenceMeanDiff"]], "conf.int" = TRUE)
+    wilxoconWCentral <- stats::wilcox.test(dat, "alternative" = direction, "mu" = 0,
+                                           "conf.level" = optionsList[["percentConfidenceMeanDiff"]], 
+                                           "conf.int" = FALSE)[["statistic"]]
+    
+    df   <- ifelse(is.null(tempResult[["parameter"]]), "", as.numeric(tempResult[["parameter"]]))
```

Comment:
since you're improving the code anyway you might as well also change this:
```suggestion
    df   <- if (is.null(tempResult[["parameter"]])) "" else as.numeric(tempResult[["parameter"]])
```

Reply:
Thanks